client/startup/i18n.js

```diff
@@ -56,29 +59,17 @@ Meteor.startup(() => {
 		loadMomentLocale(language).then(locale => moment.locale(locale), error => console.error(error));
 	};
 
-	const setLanguage = language => {
-		currentLanguage.set(filterLanguage(language));
-		localStorage.setItem('userLanguage', currentLanguage.get());
+	window.setLanguage = (language) => {
+		const lang = filterLanguage(language);
+		currentLanguage.set(lang);
+		localStorage.setItem('userLanguage', lang);
 	};
 
-	window.setLanguage = setLanguage;
-
 	window.defaultUserLanguage = () => RocketChat.settings.get('Language') || getBrowserLanguage() || 'en';
 
 	Tracker.autorun(() => {
-		const user = RocketChat.models.Users.findOne(Meteor.userId(), { fields: { username: 1 }});
-		const userLanguage = user && user.language;
-
-		const defaultLanguage = userLanguage || RocketChat.settings.get('Language') || 'en';
-
-		if (!currentLanguage.get()) {
-			setLanguage(defaultLanguage);
-		}
-
-		if (userLanguage && userLanguage !== currentLanguage.get()) {
-			setLanguage(userLanguage);
-		}
```

Comment:
As a side effect of not invoking `setLanguage()`, the reactive `currentLanguage` variable is not changed and the form login will always show a suggested language. When nor user and admin language are set, e.g., the `English version` link is always present.

Reply:
actually `setLanguage` is still being called by the login form: https://github.com/RocketChat/Rocket.Chat/blob/8bfa5988068b7dae1a2ce16dca487a9dfa7a4a6d/packages/rocketchat-ui-login/client/login/footer.js#L33

this is the only place that still calls it.